scripts/get_indexfiles.py

```diff
@@ -2,88 +2,134 @@
+                        choices=['debug', 'info', 'warning', 'error', 'critical'],
+                        help='the logging level.')
+    args = parser.parse_args()
+
+    args.field_list = {field.strip() for field in args.field_list.split(',')}
+    return args
 
 
 def main():
     args = parse_arguments()
-    i = 0
-    log_file = args.log_file + '.{}'.format(i)
-    rcode = download_index(args.query, args.output_dir,
-                           '-c {}'.format(args.collection), log_file)
-    while rcode == 0 and i <= args.max_retry:
-        uncompleted = get_uncompleted(log_file)
-        if not uncompleted:
-            break
-        i += 1
-        log_file = args.log_file + '.{}'.format(i)
-        print('Doing {}th round of full retry'.format(i), file=sys.stderr)
-        if os.path.exists(log_file):
-            os.remove(log_file)
-        for coll, pages in uncompleted:
-            time.sleep(30)  # Sleep to prevent DDoS
-            download_index(args.query, args.output_dir,
-                           '-c {} --pages {}'.format(coll, ' '.join(pages)),
-                           log_file, 'a')
-
-    if i <= args.max_retry:
-        print('Successfully finished after {} iterations!'.format(i + 1),
-              file=sys.stderr)
+
+    logging.basicConfig(
+        level=getattr(logging, args.log_level.upper()),
+        format='%(asctime)s - %(process)s - %(levelname)s - %(message)s'
+    )
+
+    base_url = 'https://data.commoncrawl.org/cc-index/collections/' \
+               f'{args.collection}/indexes/'
+
+    if args.clusters_dir:
+        clusters_dir = args.clusters_dir
+        logging.info(f'Using clusters directory {clusters_dir}.')
     else:
-        print('Finished after {} of {} iterations please check if everything '
-              'is all right!'.format(i + 1, args.max_retry + 1), file=sys.stderr)
+        clusters_dir = Path(mkdtemp())
+        logging.info(f'Created temporary clusters directory {clusters_dir}.')
+
+    ensure_dir(args.output_dir)
+    ensure_dir(clusters_dir)
+
+    try:
+        # First, let's download the cluster.idx file.
+        cluster_idx = clusters_dir / f'{args.collection}_cluster.idx'
+        if not cluster_idx.is_file():
+            logging.info(f'Downloading cluster index for {args.collection}...')
+            urllib.request.urlretrieve(base_url + 'cluster.idx', cluster_idx)
+
+        # Then, get the files and byte ranges that correspond to the TLD
+        clusters = find_tld_in_index(args.tld, cluster_idx)
+        logging.info(f'Found {len(clusters)} clusters to download.')
+        tldp = re.compile(f'^{args.tld},')
+
+        with closing(BatchWriter(
+            args.lines_per_file, args.output_dir,
+            num_digits(len(clusters) * CLUSTER_SIZE // args.lines_per_file + 1),
+            f'domain-hu-{args.collection}-'
+        )) as bw:
+            for frange in ranges_from_clusters(clusters, args.batch_size):
+                time.sleep(args.delay)
```

Comment:
Why sleep here, at the start of the current iteration, and not at the end of it?

Reply:
What does it matter? It's still one unnecessary sleep either way. Also, if we had to download `clusters.idx`, sleeping at the beginning is better because we put a delay between that and the first index file.

I'd understand if you had an issue with the one extra sleep, but since we are talking about one second, it's something we can afford, I think.